crates/accelerate/src/target_transpiler/instruction_properties.rs

```diff
@@ -0,0 +1,78 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::{prelude::*, pyclass};
+
+/**
+ A representation of the properties of a gate implementation.
+
+This class provides the optional properties that a backend can provide
+about an instruction. These represent the set that the transpiler can
+currently work with if present. However, if your backend provides additional
+properties for instructions you should subclass this to add additional
+custom attributes for those custom/additional properties by the backend.
+*/
+#[pyclass(subclass, module = "qiskit._accelerate.target")]
+#[derive(Clone, Debug)]
+pub struct BaseInstructionProperties {
```

Comment:
A suggestion on naming, from Rust space I think we should call this `InstructionProperties` because it's a rust representation of the instruction properties, there isn't anything building off of it in rust space so `Base` is more confusing. We can set a python space name in the `pyclass` macro to make it `BaseInstructionProperties` for python where the name makes more sense.

```suggestion
#[pyclass(subclass, name = "BaseInstructionProperties", module = "qiskit._accelerate.target")]
#[derive(Clone, Debug)]
pub struct InstructionProperties {
```